rotkehlchen/tasks/manager.py

```diff
@@ -547,13 +557,20 @@ def _schedule(self) -> None:
         if not_proceed:
             return  # too busy
 
-        callables = random.sample(
-            population=self.potential_tasks,
-            k=min(self.max_tasks_num - current_greenlets, len(self.potential_tasks)),
-        )
+        random.shuffle(self.potential_tasks)
+        max_tasks = min(self.max_tasks_num - current_greenlets, len(self.potential_tasks))
 
-        for callable_fn in callables:
-            callable_fn()
+        spawned_new = 0
+        for callable_fn in self.potential_tasks:
+            if spawned_new >= max_tasks:
+                break
+            if callable_fn in self.running_greenlets:
+                continue
+            new_greenlet = callable_fn()
+            if new_greenlet is None:
+                continue
```

Comment:
Add some comments for readability and for all of us to easily remember in the future what is happening. Probably last comment is the most important to have.

```suggestion
            if spawned_new >= max_tasks:
                break  # no more task slots left
            if callable_fn in self.running_greenlets:
                continue  # the specified task is already running
            new_greenlet = scheduling_fn()
            if new_greenlet is None:
                continue  # The scheduling function for the specific task decided to not schedule it
```